worker.go

```diff
@@ -54,29 +61,52 @@ func (w *CeleryWorker) StartWorker() {
 						continue
 					}
 
-					//log.Printf("WORKER %d task message received: %v\n", workerID, taskMessage)
-
 					// run task
 					resultMsg, err := w.RunTask(taskMessage)
-					if err != nil {
-						log.Println(err)
+					if err == nil {
+						// happy path
+						w.storeResult(taskMessage.ID, resultMsg)
+
+						// release the result resources
+						releaseResultMessage(resultMsg)
+						continue
+					}
+
+					taskMessage.Retries++
+					if err != ErrTaskRetryable || taskMessage.Retries >= taskMessage.Settings.MaxRetries {
```

Comment:
There is a small problem with the retry count right? because its incremented for the first try which fails which is not technically a retry. So the way to counter that is to have the condition as `taskMessage.Retries > taskMessage.Settings.MaxRetries`
Also may be we can name `taskMessage.Retries` as just `taskMessage.Tries`
So the final condition becomes
`taskMessage.Tries > taskMessage.Settings.MaxRetries` or more specifically(nicer to read) `taskMessage.Tries - 1 == taskMessage.Settings.MaxRetries`

Reply:
I dont think so. Can you take a look at tests I have written for this pls

Reply:
I think the test assumes the same behavior as the implementation, I commented there as well
Anyway we can resolve this just by 
`taskMessage.Retries++` as the last line in this block (after `w.broker.SendCeleryMessage(getCeleryMessage(enc))`)
Then it will allow for numRetries + 1 attempts of the task. First attempt is not a retry.

@mikiquantum could you also check? may be I'm getting the logic wrong here.

Reply:
A worker shouldn't assume if it is the first time or second time it is running the task but instead should run the task and increment the retries(can be called as tries I guess). In that case, the logic is given above and tests are working as per the logic.